development/kaniko-build/README.md

```diff
@@ -0,0 +1,79 @@
+# kaniko-build
+
+This tool serves as an intelligent wrapper for `kaniko-project/executor`. It reduces the complexity of building Docker images and removes the need of using Docker in Docker when building images in K8s infrastructure.
+
+Key features:
+* automatically provides a default tag, which is computed based on a template provided in `config.yaml`
+* ~~allows for concurrent builds of image variants that use the same `Dockerfile`~~ See [Known issues](#known-issues) #1
+* supports adding multiple tags to the image
+* saves the `kaniko-project/executor` command outputs to separate files
+* when running in Prow's presubmit job, supports pushing images to different repositories with different tags 
+* supports pushing the same images to multiple repositories
+* supports caching of built layers to reduce build times
+
+## Known issues
+
+1. Currently, building different variants of the same image is not working. The issue is tracked in https://github.com/kyma-project/test-infra/issues/5975
+2. This tool is still at an early stage of development. It is stable enough as a replacement for `docker build`. However, you can expect bugs and codebase changes.
+
+For any other problems, please raise an [issue](https://github.com/kyma-project/test-infra/issues/new?assignees=&labels=area%2Fci%2C+bug&template=bug-report.md&title=kaniko-build:%20).
+
+## Use config.yaml file
+
+`kaniko-build` requires a configuration file to be provided with a set of variables, which are used during the execution.
+A `--config` flag is required.
+
+For more information, refer to the [config.go](./config.go) file.
+
+Example file:
+```yaml
+registry: eu.gcr.io/kyma-project
+reproducible: true
+log-format: json
+cache:
+  enabled: true
+  cache-repo: eu.gcr.io/sap-kyma-neighbors-dev/cache
+  cache-run-layers: true
+```
+
+## Build multiple variants of the same image
+
+The tool also includes the functionality of re-using the same `Dockerfile` to concurrently build different variants of the same image.
+`variants.yaml` is the file that contains pre-defined set of the same `ARG` substitutions with different values.
+It's very handy when there is a requirement to build an image with different versions of the same binary, for example, for different versions of Kubernetes or Go.
+
+The file has a simple structure:
+```yaml
+'main':
+  KUBECTL_VERSION: "1.24.4"
+'1.23':
+  KUBECTL_VERSION: "1.23.9"
+```
+
+To use this feature, make sure that:
```

Comment:
This looks like a prerequisite for using the `variants.yaml` file. If yes, it needs to go in front of the instructions how to use that, because first the prerequisites need to be fulfilled before users can start using it.

So, for example, after line 41 (just guessing), I'd slip in:

> To use this feature, fulfill the following prerequisites:
> 
> * The `variants.yaml` file is stored in the **same directory** as the `Dockerfile`.
> * Your `Dockerfile` contains `ARG` directives that are named after keys in `variants.yaml`.

Reply:
I think that this place is also fine. Because first line is the description of the functionality. I've formatted it as simple paragraph, rather than prerequisites in mind.